scripts/background/AudioManager.js

```diff
@@ -248,7 +248,9 @@ function AudioManager(addEventListener, isTownTune) {
 
 	addEventListener("gameChange", playHourlyMusic);
 
-	addEventListener("weatherChange", playHourlyMusic);
+	addEventListener("weatherChange", playHourlyMusic, () => {
+		fadeOutAudio(300);
+	});
```

Comment:
You can't have multiple callbacks in `addEventListener` as defined [here](https://github.com/animal-crossing-music-extension/ac-music-extension/blob/f0303b8410f8d5a7092a036ac2c998b0069bb1ef/scripts/background/StateManager.js#L22-L25).
And I'm not sure this is a good idea to fade out the audio while changing it.

```suggestion
	addEventListener("weatherChange", () => {
	        fadeOutAudio(300);
	        setTimeout(playHourlyMusic, 300);
	});
```

Reply:
Ah thank you, I was not familiar with eventListener works and it seemed to fix it but this is probably a better way.